### PR TITLE
delete COS bucket during teardown of cluster

### PR DESCRIPTION
**Pull request by vavuthu**
Description:

Currently we are leaving COS buckets in IBM cloud as leftovers which was created by noobaa. So we are fetching bucket from backingstore and deleting.

Signed-off-by: vavuthu [vavuthu@redhat.com](mailto:vavuthu@redhat.com)

### CODE REVIEW

**Comment by vavuthu** [comment]
verified: https://url.corp.redhat.com/26fdc79

logs

```
2024-12-05 21:47:16  11:17:16 - MainThread - /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/ocs_ci/utility/ibmcloud.py - INFO  - Deleting bucket: nb.1733404663133.apps.vavuthuibmt3.*.com
2024-12-05 21:47:16  11:17:16 - MainThread - /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/ocs_ci/utility/ibmcloud.py - INFO  - Retrieving bucket contents from nb.1733404663133.apps.vavuthuibmt3.*.com
2024-12-05 21:47:17  11:17:16 - MainThread - /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/ocs_ci/utility/ibmcloud.py - INFO  - bucket objects: []
2024-12-05 21:47:17  11:17:17 - MainThread - /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/ocs_ci/utility/ibmcloud.py - INFO  - Bucket: nb.1733404663133.apps.vavuthuibmt3.*.com deleted!
```

File: ocs_ci/utility/ibmcloud.py [hunk 1]
```diff
@@ -1237,3 +1240,112 @@ def find_free_network_subnets(subnet_cidr, network_prefix=27):
...
+        """
+        Initialize all necessary parameters
+
+        Args:
+            api_key (str): API key for IBM Cloud Object Storage (COS)
+            service_instance_id (str): Service instance ID for COS
+            endpoint_url (str): COS endpoint URL
+
+        """
+        self.COS_API_KEY_ID = api_key
```
Comment (petr-balogh) [inline ocs_ci/utility/ibmcloud.py:1260]: Any reason for upper case name of vars?  This is usually used in constants, which is not the case here I think.

Comment (vavuthu) [reply ocs_ci/utility/ibmcloud.py]: chnaged to lowercase

**Comment by vavuthu** [comment]
re-verified: https://url.corp.redhat.com/c2281f7

**Review by petr-balogh — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Comment by petr-balogh** [comment]
@vavuthu please cherrypick down to 4.14

**Pull request merged**